lib/ui/text.dart

```diff
@@ -3253,7 +3253,24 @@ class Paragraph extends NativeFieldWrapperClass1 {
   /// metrics, so use it sparingly.
   TextRange getLineBoundary(TextPosition position) {
     final List<int> boundary = _getLineBoundary(position.offset);
-    return TextRange(start: boundary[0], end: boundary[1]);
+    final TextRange line = TextRange(start: boundary[0], end: boundary[1]);
+
+    // _getLineBoundary only considers the offset and assumes that the
+    // TextAffinity is upstream. In the case that TextPosition is just after a
+    // wordwrap (downstream), we need to get the line for the next offset.
```

Comment:
If the line ends with a RTL run then the `_getLineBoundary` function will still report the same line for `position.offset + 1` I think?

Reply:
Yeah that's what I'm seeing, but I'm glad you brought this up.  I just took a dive into Flutter RTL keyboard support while investigating this and it seems very broken.  I don't know anything about any RTL languages unfortunately, but the behavior is very different from the browser.  I will probably create a new issue for that.